Sources/Core/Screen/ScreenConnectionMonitor.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  ScreenConnectionMonitor.swift
+//  XestiMonitors
+//
+//  Created by Paul Nyondo on 2018-04-04.
+//
+//  © 2018 J. G. Pusey (see LICENSE.md)
+//
+
+#if os(iOS) || os(tvOS)
+
+import UIKit
+
+///
+/// A `ScreenConnectionMonitor` instance monitors a device for when a new
+/// screen is connected or disconnected
+///
+public class ScreenConnectionMonitor: BaseNotificationMonitor {
+    ///
+    /// Encapsulates changes when a user connects or disconnects
+    /// a screen
+    ///
+    public enum Event {
+        ///
+        /// A new screen is connected to the device.
+        ///
+
+        case didConnect(UIScreen)
+
+        ///
+        /// The screen is disconnected from the device.
+        ///
+
+        case didDisconnect(UIScreen)
+    }
+
+    ///
+    /// Specifies which events to monitor.
+    ///
+    public struct Options: OptionSet {
+        ///
+        /// Monitor `didConnect` events.
+        ///
+        public static let didConnect = Options(rawValue: 1 << 0)
+
+        ///
+        /// Monitor `didDisconnect` events.
+        ///
+
+        public static let didDisconnect = Options(rawValue: 1 << 1)
+
+        ///
+        /// Monitor all events.
+        ///
+        public static let all: Options = [.didConnect,
+                                          .didDisconnect]
+        /// :nodoc:
+        public init(rawValue: UInt) {
+            self.rawValue = rawValue
+        }
+
+        /// :nodoc:
+        public let rawValue: UInt
+    }
+
+    ///
+    /// Initializes a new `ScreenConnectionMonitor`.
+    ///
+    /// - Parameters:
+    ///   - screen:     The screen to monitor.
```

Comment:
It seems to me that this monitor should be monitoring _all_ screens. I don't think the user will actually have a `UIScreen` instance for a screen that has not yet been connected. So I think the `screen` parameter and property should be eliminated. The correct `UIScreen` instance will still be found in `notification.object`.